### PR TITLE
[QMS-624] Suspicion: QMS crashes if BRouter RD5 file is missing

### PR DESCRIPTION
**Pull request by kiozen**
### What is the linked issue for this pull request:
[comment]: # (Add the ticket number behind QMS-# )

QMS-#624

### What you have done:
[comment]: # (Describe roughly.)

Do not forward the exception in `CRouterBRouter::synchronousRequest()` Display the error on the canvas

### Steps to perform a simple smoke test:
[comment]: # (List the steps.)

Follow the descriptions in the ticket

### Does the code comply to the coding rules and naming conventions [Coding Guidelines](https://github.com/Maproom/qmapshack/wiki/DeveloperCodingGuideline):

- [x] yes

### Is every user facing string in a tr() macro?

- [x] yes

### Did you add the ticket number and title into the changelog? Keep the numeric order in each release block.

- [x] yes, I didn't forget to change changelog.txt

### CODE REVIEW

File: src/qmapshack/gis/rte/router/CRouterBRouter.cpp [hunk 1]
```diff
@@ -383,9 +383,12 @@ int CRouterBRouter::synchronousRequest(const QVector<QPointF>& points, const QLi
   } catch (const QString& msg) {
     coords.clear();
     if (!msg.isEmpty()) {
+      CCanvas* canvas = CMainWindow::self().getVisibleCanvas();
+      if (canvas) {
+        canvas->reportStatus("BRouter", tr("<b>BRouter</b><br/>Bad response from server:<br/>%1").arg(msg));
```
Comment (ntruchsess) [inline src/qmapshack/gis/rte/router/CRouterBRouter.cpp:388]: this status-message cannot be seen as it is closed immediately by the call of 'slotCloseStatusMsg()' in line 396

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/qmapshack/gis/rte/router/CRouterBRouter.cpp (hunk 1 above)
Comment (kiozen) [reply src/qmapshack/gis/rte/router/CRouterBRouter.cpp]: You got a point here.  In the meanwhile I could follow the call paths and I am pretty confident that the exceptions are caught while routing on-the-fly. 

I added a few `noexcept(false)`  for readability. Else they are worth nothing.

**Review by ntruchsess — CHANGES_REQUESTED** [review]
looks good, please incorporate my suggested changes of https://github.com/kiozen/qmapshack/pull/1

**Comment by kiozen** [comment]
> looks good, please incorporate my suggested changes of [kiozen#1](https://github.com/kiozen/qmapshack/pull/1)

Done.

And now we still need to know why it crashes.

**Pull request closed without merging**